src/main/java/com/petros/bibernate/dao/EntityPersister.java

```diff
@@ -146,6 +144,50 @@ public <T> T update(T entity) {
         }
     }
 
+    /**
+     * Deletes an existing entity from the database.
+     *
+     * @param entity the entity to delete
+     * @param <T>    the type of the entity
+     * @return the deleted entity
+     */
+    public <T> T delete(T entity) {
+        Objects.requireNonNull(entity);
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement deleteStatement = prepareDeleteStatement(entity, connection)) {
+            int rowsAffected = deleteStatement.executeUpdate();
+            throwExceptionIfRowsAffectedNotOne(rowsAffected, "Failed to delete entity from the database");
+            return entity;
+
+        } catch (SQLException | IllegalAccessException e) {
+            throw new BibernateException(e);
+        }
+    }
+
+    private static <T> PreparedStatement prepareFindStatement(Class<T> entityClass, Field field, Object fieldValue,
+                                                              Connection connection) throws SQLException {
+        String tableName = getTableName(entityClass);
+        String columnName = getColumnName(field);
+        String query = String.format(FIND_ENTITY_BY_FIELD_NAME_TEMPLATE, tableName, columnName);
+        PreparedStatement statement = connection.prepareStatement(query);
```

Comment:
You moved prepared statement creation from try-with-resources, so statement is not closed automatically. Please, add statement.close() to each of CRUD methods.

Reply:
In all CRUD operations, the prepareCrudOperationStatement method is called inside a try-with-resources block. Therefore, all prepared statements will be closed automatically

Reply:
Cannot find something like `try (var st = connection.createStatement()) {...}`
Or you meant that every autocloseable object inside curly brackets will be closed automatically, without mentioning inside round brackets?